sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs

```diff
@@ -299,10 +299,16 @@ public partial class TokenCacheUpdatedArgs
         internal TokenCacheUpdatedArgs() { }
         public System.ReadOnlyMemory<byte> UnsafeCacheData { get { throw null; } }
     }
+    public partial class TokenCredentialDiagnosticsOptions : Azure.Core.DiagnosticsOptions
+    {
+        public TokenCredentialDiagnosticsOptions() { }
+        public bool IsAccountIdentifierLoggingEnabled { get { throw null; } set { } }
+    }
     public partial class TokenCredentialOptions : Azure.Core.ClientOptions
     {
         public TokenCredentialOptions() { }
         public System.Uri AuthorityHost { get { throw null; } set { } }
+        public new Azure.Identity.TokenCredentialDiagnosticsOptions Diagnostics { get { throw null; } }
```

Comment:
Did we consider making the Diagnostics property on ClientOptions virtual?

Reply:
We did consider that, but this doesn't work as we can't change the return type when overriding while targitting netstandard2.0. After prototyping several different ways to override the `DiagnosticsOptions` and `RetryOptions` we ended up preferring this design as it keeps the logic of copying default values from `ClientOptions.Default`, in the `Azure.Core` library and doesn't introduce the possibility of derived types not honoring the static defaults. Also, in this design the base class is still responsible for storing the `Diagnostics` options as well so we don't end up creating duplicate instances. All the new property here does is cast the base property to the more specific type `TokenCredentialDiagnosticsOptions`, so there's no harm in it not being respected when a user down casts the `TokenCredentialOptions` type to `ClientOptions`. The same instance will be returned just as the base type, `DiagnosticsOptions`.